Improve highlighting and coverage (#63)
nix-community/nix-ts-mode: README.org

[[https://stable.melpa.org/#/nix-ts-mode][file:https://stable.melpa.org/packages/nix-ts-mode-badge.svg]]
[[https://github.com/nix-community/nix-ts-mode/actions/workflows/trunk.yaml][https://img.shields.io/github/actions/workflow/status/nix-community/nix-ts-mode/trunk.yaml.svg?label=Trunk&event=push&branch=trunk]]

An Emacs major mode for editing Nix expressions, powered by the built-in

> tree-sitter support in Emacs 30+ and the community-maintained [[https://github.com/nix-community/tree-sitter-nix][Nix tree-sitter grammar]].

Reviewer: ```suggestion
tree-sitter support in Emacs 29+ and the community-maintained [[https://github.com/nix-community/tree-sitter-nix][Nix tree-sitter grammar]].
```
`tree-sitter` support in Emacs 29 is still there. 😄

*Features:*
- Improved syntax highlighting for all Nix language constructs
- Semantic distinctions (variables vs properties, function calls vs definitions)
- 100% tree-sitter-nix grammar coverage
- Compatible with nixd LSP server

> - 31 comprehensive tests

Reviewer: ```suggestion
- Syntax highlighting for all Nix language constructs
- Semantic distinctions (variables vs properties, function calls vs definitions)
- 100% `tree-sitter-nix` grammar coverage
- Comprehensive tests
```
Keeping this minimal and not prone to getting out of sync.

    (use-package nix-ts-mode
     :mode "\\.nix\\'")
#+END_SRC

*** Maximum Highlighting

> For the best syntax highlighting experience with all 19 font-lock faces, set the font-lock level to 4 (maximum):

Reviewer: ```suggestion
For the best syntax highlighting experience with the widest selection of font-lock faces, set the font-lock level to 4 (maximum):
```

- ✅ font-lock-constant-face - true/false/null
- ✅ font-lock-delimiter-face - commas, semicolons
- ✅ font-lock-escape-face - \n, \t, ''$
- ✅ font-lock-keyword-face - let/in/if/then/inherit
- ✅ font-lock-number-face - integers, floats
- ✅ font-lock-operator-face - +, -, ?, @, =
- ✅ font-lock-property-name-face - binding names
- ✅ font-lock-property-use-face - property access
- ✅ font-lock-punctuation-face - ${}, quotes
- ✅ font-lock-string-face - strings, paths
- ✅ font-lock-type-face - base objects (semantic)
- ✅ font-lock-variable-name-face - parameters
- ✅ font-lock-variable-use-face - variables
- ✅ font-lock-warning-face - deprecated constructs
- ✅ font-lock-function-call-face - function calls
- ✅ font-lock-function-name-face - function definitions

*** Faces NOT Used (Not applicable to Nix)

- font-lock-comment-delimiter-face - (we use comment-face for whole line)
- font-lock-doc-face / font-lock-doc-markup-face - (Nix has no doc comments)
- font-lock-misc-punctuation-face - (could be used as micro-refinement for : and ?)
- font-lock-negation-char-face - (Nix uses ! as operator, not negation char)
- font-lock-preprocessor-face - (no preprocessor in Nix)

> - font-lock-regexp-face - (no regex literals in Nix)

Reviewer: ```suggestion
```

> - font-lock-regexp-face - (no regex literals in Nix)

Reviewer: No need to detail these in documentation and subsequently maintain it.
```suggestion
```
